### DISCUSSION TITLE
DAB-Tuner not Found

### DISCUSSION THREAD

**Discussion by hjfisch** [rern/rAudio#1783] [Bug Reports]
![Screenshot 2024-09-23 131836](https://github.com/user-attachments/assets/463b986a-3c1f-4555-8888-558adcedc3cb)
![Screenshot 2024-09-23 131925](https://github.com/user-attachments/assets/b6770f63-3471-4238-93a3-e330111bba08)
DAB Tuner seems to be found , but telling me Not Found

**Comment by rern** [comment]
Please try: `sed -i 's/timeout 1/timeout 3/' /srv/http/bash/settings/features.sh`

**Comment by hjfisch** [comment]
![Screenshot 2024-09-23 160905](https://github.com/user-attachments/assets/058eed9f-7cf4-4a06-92ea-1c04666232a6)
Still the same : No DAB device found , but now I cannot switch-on the device . Before I was not able to switch-off

**Reply by kan-ibal** [reply]
> but now I cannot switch-on the device . Before I was not able to switch-off

It looks like it crashes. 
What version of RPI are you using? DAB player is useless for RPI version <=3 due to lack of CPU computation power. See another thread.

**Comment by hjfisch** [comment]
Raspberry Pi 4B Rev 1.5

**Reply by kan-ibal** [reply]
Please, see if you can find any station using:
`/bin/dab-scanner-rtlsdr -C 5A`

**Comment by hjfisch** [comment]
No stations . I've tried two dab-sticks now with the same result.
One of them was running last year , but I do not know with which version of rAudio .
I

**Comment by kan-ibal** [comment]
DAB reception is heavily rely on signal strength from antenna so strongly recommend to check this. I have several stations that are detected but unable to play.

**Comment by hjfisch** [comment]
I made a new sdcard with the latest build and I'am not able to switch DAB on and yes I've controlled my antenna which is still the same
I used before . The stick seems to be detected , but rAudio says : No DAB devices found . Maybe the plugin needs an update and some libraries are not compatible with the latest versions .

**Comment by kan-ibal** [comment]
Should `/bin/dab-scanner-rtlsdr -C 5A` works it means that is detected.

**Comment by hjfisch** [comment]
![Screenshot 2024-09-24 102247](https://github.com/user-attachments/assets/6304fbf0-8d17-4ffb-a153-a05bbde054ae)
A little screenshot .
It detects a lot of stations . The screenshot shows only some .

**Reply by kan-ibal** [reply]
It looks good. So if I see that you have E4000 tuner and it is not working but it works for me with no E4000 tuner then you can try:
`sed -i 's/rtl_test -t/rtl_test/' /srv/http/bash/settings/features.sh `

**Comment by hjfisch** [comment]
Nothing has changed . I have a stick with no E4000 tuner too , but it is still the same.

**Reply by kan-ibal** [reply]
I can read Elonics E4000 tuner from screenshots.

**Comment by hjfisch** [comment]
![Screenshot 2](https://github.com/user-attachments/assets/2896c23b-c1aa-4bb7-82de-705d34f6846a)
Yes , but I have 2 different Sticks .
I've changed it now and this screenshot is from the one without E4000

**Comment by hjfisch** [comment]
I've testet now with a RPI 3B+ and still no DAB Device found

**Comment by kan-ibal** [comment]
OMG! Tinkering with rtl_test in /srv/http/bash/settings/features.sh I ended up with the same "No DAB devices found" but I can search for stations in Library/DAB and play it. It looks like PHP sees it as disable only.
Use your no E400 dongle, revert to rtl_test with -t and go Library/DAB to see if you can find stations.

**Reply by kan-ibal** [reply]
> It looks like PHP sees it as disable only.
Not directly related but `$id_data` table from `/srv/http/settings/features.php` looks odd for upmpdcli
`'upmpdcli'       => [ 'label' => 'UPnP / DLNA',      'sub' => 'upmpdcli',       'setting' => false,    'status'  => true, 'exist' => 'spotifyd' ]`
there should be `'exist' => 'upmpdcli'`  instead of `'exist' => 'spotifyd' `

**Reply by kan-ibal** [reply]
There is `$( '#dabradio' ).toggleClass( 'disabled', ! S.dabdevice )` in `/srv/http/assets/js/features.js` but
`dabdevice` variable is missing in data of `/srv/http/bash/settings/features-data.sh`
e.g.
`, "dabdevice"        : '$( timeout 1 rtl_test -t &> /dev/null && echo true )'`

**Comment by hjfisch** [comment]
Due to the fact that I cannot switch-on DAB , there is no DAB-Icon in Library . In /srv/http/data I can find a directory called dabradio
![Screenshot 3](https://github.com/user-attachments/assets/54ba021a-fece-419e-ac84-d08053daaacc)
Now rAudio does not see my DAB-Sick anymore.

**Comment by hjfisch** [comment]
Got it working now with rAudio image rAudio 20231022 .
So it seems to be a bug in in latest rAudio version .
Tested both Sticks and both are  working with this version .

**Comment by hjfisch** [comment]
Due to the fact that there are to many bugs in rAudio20231022 , I need a fix to get it work in the latest build .

**Reply by kan-ibal** [reply]
I have already provided the solution:
`sed -i '/, "dabradio"         : '$mediamtx'/a , "dabdevice"        : '\''\$\( timeout 1 rtl_test -t &> /dev/null && echo true \)'\''' /srv/http/bash/settings/features-data.sh`

**Comment by hjfisch** [comment]
Hello ,
I'm sorry , but your solution does not work for me . Still the same "No DAB Device found" .
Only with some old versions of rAudio from last year  it is working .

**Comment by hjfisch** [comment]
Now DAB-plugin could not be installed .
"error: failed retrieving file 'extra.db' from mirror.archlinuxarm.org : Operation too slow. Less than 1 bytes/sec transferred the last 10 seconds
error: failed to synchronize all databases (download library error)"

**Comment by hjfisch** [comment]
A few minutes later it works .
"Found 1 device(s):
  0:  Nooelec, SMArt XTR v5, SN: 00000001

Using device 0: Generic RTL2832U OEM
Detached kernel driver
Found Elonics E4000 tuner
Supported gain values (14): -1.0 1.5 4.0 6.5 9.0 11.5 14.0 16.5 19.0 21.5 24.0 29.0 34.0 42.0 
Sampling at 2048000 S/s.
Benchmarking E4000 PLL...
Signal caught, exiting!
[E4K] PLL not locked for 51000000 Hz!
[E4K] PLL not locked for 2177000000 Hz!
[E4K] PLL not locked for 1090000000 Hz!
[E4K] PLL not locked for 1260000000 Hz!
E4K range: 52 to 2176 MHz
E4K L-band gap: 1090 to 1260 MHz
Reattached kernel driver

# systemctl status mediamtx
* mediamtx.service - MediaMTX
     Loaded: loaded (/usr/lib/systemd/system/mediamtx.service; enabled; preset: disabled)
     Active: active (running) since Fri 2024-10-25 09:26:49 CEST; 9min ago
 Invocation: d8047dd3d60e442c99c1e7988f838d40
   Main PID: 15560 (mediamtx)
      Tasks: 8 (limit: 4465)
        CPU: 173ms
     CGroup: /system.slice/mediamtx.service
             └─15560 /usr/bin/mediamtx /etc/mediamtx/mediamtx.yml

Oct 25 09:26:49 hjfSound systemd[1]: Started MediaMTX.
Oct 25 09:26:49 hjfSound mediamtx[15560]: 2024/10/25 09:26:49 INF MediaMTX v1.9.3
Oct 25 09:26:49 hjfSound mediamtx[15560]: 2024/10/25 09:26:49 INF configuration loaded from /etc/mediamtx/mediamtx.yml
Oct 25 09:26:49 hjfSound mediamtx[15560]: 2024/10/25 09:26:49 INF [RTSP] listener opened on :8554 (TCP), :8000 (UDP/RTP), :8001 (UDP/RTCP)
Oct 25 09:26:49 hjfSound mediamtx[15560]: 2024/10/25 09:26:49 INF [RTMP] listener opened on :1935
Oct 25 09:26:49 hjfSound mediamtx[15560]: 2024/10/25 09:26:49 INF [HLS] listener opened on :8888
Oct 25 09:26:49 hjfSound mediamtx[15560]: 2024/10/25 09:26:49 INF [WebRTC] listener opened on :8889 (HTTP), :8189 (ICE/UDP)
Oct 25 09:26:49 hjfSound mediamtx[15560]: 2024/10/25 09:26:49 INF [SRT] listener opened on :8890 (UDP)"

 DAB Icon is greyed out . Scanning Stations seems not to work .
![Screenshot 2024-10-25 093845](https://github.com/user-attachments/assets/cf3a9af6-701e-48b9-837e-289d30fdf6bf)
